### PR TITLE
[Merged by Bors] - v2alpha1: support ineffectual transaction state

### LINKED ISSUE #6466: v2alpha1: Support INEFFECTUAL transaction state
## Description

https://github.com/spacemeshos/go-spacemesh/pull/6443

This PR introduced new transaction state `TRANSACTION_STATE_INEFFECTUAL`, which is currently not supported by v2alpha1 api.

### PR DESCRIPTION
**Pull request by kacpersaw**
## Motivation

#6443 Introduced new transaction state `TRANSACTION_STATE_INEFFECTUAL` which was not supported by v2alpha1.

Closes #6466

### CODE REVIEW

File: api/grpcserver/v2alpha1/transaction_test.go [hunk 1]
```diff
@@ -757,3 +757,40 @@ func TestToTxContents(t *testing.T) {
...
+	signer, err := signing.NewEdSigner()
+	require.NoError(t, err)
+	tx := newTx(0, types.Address{}, signer)
+	require.NoError(t, db.WithTxImmediate(context.Background(), func(dtx sql.Transaction) error {
+		require.NoError(t, transactions.Add(dtx, &types.Transaction{
+			RawTx:    tx.RawTx,
+			TxHeader: nil,
+		}, time.Time{}))
+		return nil
+	}))
```
Comment (fasmat) [inline api/grpcserver/v2alpha1/transaction_test.go]: No transaction needed:

```suggestion
	require.NoError(t, transactions.Add(db, &types.Transaction{
		RawTx:    tx.RawTx,
		TxHeader: nil,
	}, time.Now()))
```

File: api/grpcserver/v2alpha1/transaction.go [hunk 2]
```diff
@@ -303,7 +304,7 @@ func toTransactionOperations(filter *spacemeshv2alpha1.TransactionRequest) (buil
 	return ops, nil
 }
 
-func toTx(tx *types.MeshTransaction, result *types.TransactionResult,
+func (s *TransactionService) toTx(ctx context.Context, tx *types.MeshTransaction, result *types.TransactionResult,
 	includeResult, includeState bool,
 ) *spacemeshv2alpha1.TransactionResponse {
```
Comment (fasmat) [inline api/grpcserver/v2alpha1/transaction.go]: NIT: I'd prefer if all arguments were on their own new line 🙂 

```suggestion
func (s *TransactionService) toTx(
	ctx context.Context,
	tx *types.MeshTransaction,
	result *types.TransactionResult,
	includeResult, includeState bool,
) *spacemeshv2alpha1.TransactionResponse {
```

File: api/grpcserver/v2alpha1/transaction.go [hunk 3]
```diff
@@ -377,6 +381,18 @@ func convertTxState(tx *types.MeshTransaction) *spacemeshv2alpha1.TransactionSta
 		state := spacemeshv2alpha1.TransactionState_TRANSACTION_STATE_PROCESSED
 		return &state
 	default:
+		evicted, err := s.conState.HasEvicted(tx.ID)
+		if err != nil {
+			ctxzap.Debug(ctx, "failed to check if tx is evicted",
+				zap.String("tx_id", tx.ID.String()),
+				zap.Error(err),
+			)
+			return nil
```
Comment (fasmat) [inline api/grpcserver/v2alpha1/transaction.go]: I'm not sure if returning `nil` is correct here. The transaction then has no state - instead I think it should be something like unknown/unspecified/error - maybe `TransactionState_TRANSACTION_STATE_UNSPECIFIED`?

**Review by fasmat — APPROVED** [review]

**Comment by kacpersaw** [comment]
bors merge

**Comment by kacpersaw** [comment]
bors merge

**Comment by kacpersaw** [comment]
bors merge

**Comment by kacpersaw** [comment]
bors merge

**Comment by kacpersaw** [comment]
bors merge

**Pull request closed without merging**